evalml/tests/component_tests/test_estimators.py

```diff
@@ -21,22 +27,24 @@ def test_estimators_feature_name_with_random_ascii(X_y_binary, X_y_multi, X_y_re
         supported_problem_types = [handle_problem_types(pt) for pt in estimator_class.supported_problem_types]
         for problem_type in supported_problem_types:
             clf = helper_functions.safe_init_component_with_njobs_1(estimator_class)
-            if problem_type == ProblemTypes.BINARY:
+            if is_binary(problem_type):
                 X, y = X_y_binary
-            elif problem_type == ProblemTypes.MULTICLASS:
+            elif is_multiclass(problem_type):
                 X, y = X_y_multi
-            elif problem_type == ProblemTypes.REGRESSION:
+            elif is_regression(problem_type):
                 X, y = X_y_regression
 
             X = get_random_state(clf.random_state).random((X.shape[0], len(string.printable)))
             col_names = ['column_{}'.format(ascii_char) for ascii_char in string.printable]
             X = pd.DataFrame(X, columns=col_names)
+            assert clf.input_feature_names is None
             clf.fit(X, y)
             assert len(clf.feature_importance) == len(X.columns)
             assert not np.isnan(clf.feature_importance).all().all()
             predictions = clf.predict(X).to_series()
             assert len(predictions) == len(y)
             assert not np.isnan(predictions).all()
+            assert (clf.input_feature_names == col_names)
```

Comment:
Rather than create a whole new test, tacking it on here, where we test passing weird names to estimators :d